libgadget/timestep.c

```diff
@@ -55,7 +55,7 @@ timestep_eh_slots_fork(EIBase * event, void * userdata)
     int parent = ev->parent;
     int child = ev->child;
 
-    if(is_timebin_active(P[parent].TimeBin, All.Ti_Current)) {
+    if(is_timebin_active(P[parent].TimeBin, All.Ti_Current) && !P[parent].Swallowed) {
```

Comment:
I think you don't need this: this hook fires if a new particle is created, but a new particle is never already swallowed.